talk/expert/sfinae.tex

```diff
@@ -6,37 +6,38 @@
   \begin{block}{The main idea}
     \begin{itemize}
     \item substitution replaces template parameters with the provided arguments (types or values)
-    \item if it leads to invalid code, do not fail but try other overloads
+    \item if it leads to invalid code, do not report an error but try other overloads/specializations
     \end{itemize}
   \end{block}
   \begin{exampleblock}{Example}
     \begin{cppcode*}{}
       template <typename T>
       void f(typename T::type arg) { ... }
-
       void f(int a) { ... }
 
       f(1); // Calls void f(int)
     \end{cppcode*}
   \end{exampleblock}
-  Note : SFINAE is largely superseded by concepts in \cpp20
+  \begin{alertblock}{}
+    Note: SFINAE is largely superseded by concepts in \cpp20
+  \end{alertblock}
 \end{frame}
 
 \begin{frame}[fragile]
-  \frametitlecpp[11]{decltype}
+  \frametitlecpp[11]{\texttt{decltype}}
```

Comment:
I realize that the usage of decltype with lambdas yesterday could not be understood, as decltype was not known...
To be skipped in the advanced course, and reserved for an expert one, if any